check_library_exists fails when using static NetCDF, HDF5, etc libraries

The way `check_library_exists` works is to compile a small test program and then link it with the necessary libraries.

However, `CMAKE_REQUIRED_LIBRARIES` is set to only link to the NetCDF library, for example:

https://github.com/nco/nco/blob/0d0a9aa2e78e0db84f1f1cd5dcf05a97c987385f/CMakeLists.txt#L152

This works fine when using shared libraries because the other libraries are implicitly linked in.

However, when using static libraries it leads to undefined symbols in `CMakeFiles/CMakeError.log` because of undefined symbols for HDF5, etc, and then the reported feature `nc_inq`, `H5Z_DEFLATE`, etc are reported as not found.

Which then leads to duplicate symbols for the NetCDF functions and zlib and szip not being linked at all.

What it should be is something like this:

```
set(NETCDF_REQUIRED_LIBRARIES ${NETCDF_LIBRARY} ${HDF5_HL_LIBRARY} ${HDF5_LIBRARY} ${ZLIB_LIBRARY} ${SZIP_LIBRARY})
set(CMAKE_REQUIRED_LIBRARIES ${NETCDF_REQUIRED_LIBRARIES})
```

And the same thing for HDF5 (minus NetCDF).

Maybe with a check if the libraries are static, though maybe not necessary.

WIth this the build worked for me.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -149,7 +149,10 @@ endif()
 #check for header file netcdf_mem.h existence
 #/////////////////////////////////////////////////////////////////////////////////////
 
-set(CMAKE_REQUIRED_LIBRARIES ${NETCDF_LIBRARY})
+set(NETCDF_REQUIRED_LIBRARIES ${NETCDF_LIBRARY} ${HDF5_HL_LIBRARY} ${HDF5_LIBRARY} ${ZLIB_LIBRARY} ${SZIP_LIBRARY})
+# Replace backslashes with forward slashes because it can cause problems with Windows' paths
+string(REPLACE "\\" "/" NETCDF_REQUIRED_LIBRARIES ${NETCDF_REQUIRED_LIBRARIES})
+set(CMAKE_REQUIRED_LIBRARIES ${NETCDF_REQUIRED_LIBRARIES})
 
 check_library_exists(${NETCDF_LIBRARY} nc_rename_grp "" has_nc_rename_grp)
 if (has_nc_rename_grp)
@@ -236,7 +239,10 @@ endif()
 #which is not a requirement for the library to build successfully
 #//////////////////////////
 
-set(CMAKE_REQUIRED_LIBRARIES ${HDF5_LIBRARY})
+set(HDF5_REQUIRED_LIBRARIES ${HDF5_HL_LIBRARY} ${HDF5_LIBRARY} ${ZLIB_LIBRARY} ${SZIP_LIBRARY})
+# Replace backslashes with forward slashes because it can cause	problems with Windows' paths
+string(REPLACE "\\" "/" HDF5_REQUIRED_LIBRARIES ${HDF5_REQUIRED_LIBRARIES})
+set(CMAKE_REQUIRED_LIBRARIES ${HDF5_REQUIRED_LIBRARIES})
 
 message("-- Detecting if HDF5 library ${HDF5_LIBRARY} needs the ZLIB library...")
 check_library_exists(${HDF5_LIBRARY} H5Z_DEFLATE "" NEED_ZLIB)

